app/src/main/java/com/example/vmedvediev/ua21summerdancecamp/UI/bottomnavigation/events/EventsAdapter.kt

```diff
@@ -1,79 +1,86 @@
 package com.example.vmedvediev.ua21summerdancecamp.UI.bottomnavigation.events
 
-import android.arch.lifecycle.LiveData
 import android.content.Context
-import android.content.Intent
 import android.support.v7.widget.CardView
 import android.support.v7.widget.RecyclerView
-import android.util.Log
 import android.view.ViewGroup
 import android.widget.TextView
-import com.example.vmedvediev.ua21summerdancecamp.MyApplication
 import com.example.vmedvediev.ua21summerdancecamp.R
 import com.example.vmedvediev.ua21summerdancecamp.inflate
+import com.example.vmedvediev.ua21summerdancecamp.model.Date
 import com.example.vmedvediev.ua21summerdancecamp.model.Event
+import com.example.vmedvediev.ua21summerdancecamp.model.ListItem
 import kotlinx.android.synthetic.main.day_item.view.*
 import kotlinx.android.synthetic.main.event_item.view.*
 
-class EventsAdapter(private val context: Context, private val eventsList: LiveData<ArrayList<Event>>) :  RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class EventsAdapter(private val context: Context, private val eventsList: ArrayList<ListItem>) :
+        RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
-    private val namesOfDaysArray = MyApplication.instance.resources.getStringArray(R.array.namesOfDaysArray)
+    companion object {
+        const val EVENT_TYPE = 0
+        const val DATE_TYPE = 1
+    }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
         return when (viewType) {
-            0 -> EventViewHolder(parent)
-            2 -> DayViewHolder(parent)
+            EVENT_TYPE -> EventViewHolder(parent)
+            DATE_TYPE -> DayViewHolder(parent)
             else -> EventViewHolder(parent)
         }
     }
 
     override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        holder.itemView
         when (holder.itemViewType) {
-            0 -> {
-                Log.e("ON BIND VIEW HOLDER::::", position.toString())
-                val event = eventsList.value?.get(position)
-                val eventHolder = holder as EventViewHolder
-                eventHolder.id.text = event?.id
-                eventHolder.name.text = event?.name
-                eventHolder.card.setOnClickListener {
-                    val intent = Intent(context, EventsActivity::class.java)
-                    context.startActivity(intent)
-                }
-            }
-            2 -> {
-                Log.e("ITEM VIEW TYPE::::", position.toString())
-                val nameOfTheDay = namesOfDaysArray[position]
-                val dayHolder = holder as DayViewHolder
-                dayHolder.day.text = nameOfTheDay
-            }
+            EVENT_TYPE -> prepareEventCard(holder, position)
+            DATE_TYPE -> prepareDateCard(holder, position)
+        }
+    }
+
+    private fun prepareEventCard(holder: RecyclerView.ViewHolder, position: Int) {
+        val event = eventsList[position] as Event
```

Comment:
Try
val event: Event = eventsList[position]

Reply:
Any way it asks for casting